src/sentry/tagstore/snuba/backend.py

```diff
@@ -16,69 +17,363 @@
 from sentry.tagstore.base import TagStorage
 from sentry.utils import snuba
 
+SEEN_COLUMN = 'received'
+
+
+class ObjectWrapper(object):
+    def __init__(self, dictionary):
+        self.__dict__ = dictionary
+
 
 class SnubaTagStorage(TagStorage):
 
+    def get_time_range(self, days=90):
+        """
+        Returns the default (start, end) time range for querrying snuba.
+        """
+        end = datetime.utcnow().replace(tzinfo=pytz.UTC)
+        return (end - timedelta(days=days), end)
+
     # Tag keys and values
     def get_tag_key(self, project_id, environment_id, key, status=TagKeyStatus.VISIBLE):
-        pass
+        # TODO is status applicable?
+        return self.get_group_tag_key(project_id, None, environment_id, key)
 
     def get_tag_keys(self, project_id, environment_id, status=TagKeyStatus.VISIBLE):
-        pass
+        # TODO why is there no limit?
+        return self.get_group_tag_keys(project_id, None, environment_id)
 
     def get_tag_value(self, project_id, environment_id, key, value):
-        pass
+        return self.get_group_tag_value(project_id, None, environment_id, key, value)
 
     def get_tag_values(self, project_id, environment_id, key):
-        pass
+        return self.get_group_tag_values(project_id, None, environment_id, key)
 
     def get_group_tag_key(self, project_id, group_id, environment_id, key):
-        pass
+        from sentry.tagstore.exceptions import GroupTagKeyNotFound
+        start, end = self.get_time_range()
+        tag = 'tags[{}]'.format(key)
+        filters = {
+            'project_id': [project_id],
+            'environment': [environment_id],
+        }
+        if group_id is not None:
+            filters['issue'] = [group_id]
+        conditions = [[tag, '!=', '']]
+        aggregations = [['count', '', 'count']]
+
+        result = snuba.query(start, end, [], conditions, filters, aggregations)
+        if result == 0:
+            raise GroupTagKeyNotFound
+        else:
+            return ObjectWrapper({
+                'times_seen': result,
+                'key': key,
+                'group_id': group_id,
+            })
 
     def get_group_tag_keys(self, project_id, group_id, environment_id, limit=None):
-        pass
+        limit = limit or 1000
```

Comment:
Should this be in the method sig or was it temporary?

Reply:
fixed